server.go

```diff
@@ -352,6 +364,26 @@ func WithoutLogger() func(*Server) {
 	}
 }
 
+// WithListener configures the server to use a custom listener.
+func WithListener(listener net.Listener) func(*Server) {
+	return func(s *Server) {
+		if s.listener != nil {
+			panic("a listener is already configured; cannot overwrite it")
+		}
+		s.isTLS = isTLSListener(listener)
```

Comment:
It’s not really apparent to me why we care necessarily if the listener is TLS or not?